app/models/course.rb

```diff
@@ -78,4 +100,15 @@ def duration(format = "mins")
     Duration.minutes_str(total, format)
   end
 
+  def set_pub_date
+    pub_date = DateTime.now.strftime("%m/%d/%Y") if pub_status == "P"
```

Comment:
Found a bit of an issue with this

```
irb(main):006:0> course.pub_date = DateTime.now
=> Thu, 05 Nov 2015 12:51:32 -0700
irb(main):007:0> course.pub_date
=> Thu, 05 Nov 2015 19:51:32 UTC +00:00
irb(main):008:0> course.pub_date = DateTime.now.strftime("%m/%d/%Y")
=> "11/05/2015"
irb(main):009:0> course.pub_date
=> Mon, 11 May 2015 00:00:00 UTC +00:00
```

When doing the strftime conversion, it get's converted to the American format for date, but then Rails parses it as if it's the International standard format. This is where Rails can get confused a bit. It would be better to just store it using ```DateTime.now``` and no conversion, than to convert it. If you want it to store the beginning of the day, use ```DateTime.now.beginning_of_day``` though this is where just altering it on the front end when displayed tends to be better.